internal/notification/channels/twilio/config.go

```diff
@@ -1,11 +1,31 @@
 package twilio
 
+import (
+	newTwilio "github.com/twilio/twilio-go"
+	verify "github.com/twilio/twilio-go/rest/verify/v2"
+
+	"github.com/zitadel/zitadel/internal/zerrors"
+)
+
 type Config struct {
-	SID          string
-	Token        string
-	SenderNumber string
+	SID              string
+	Token            string
+	SenderNumber     string
+	VerifyServiceSID string
 }
 
 func (t *Config) IsValid() bool {
 	return t.SID != "" && t.Token != "" && t.SenderNumber != ""
 }
+
+func (t *Config) VerifyCode(verificationID, code string) error {
+	client := newTwilio.NewRestClientWithParams(newTwilio.ClientParams{Username: t.SID, Password: t.Token})
+	checkParams := &verify.CreateVerificationCheckParams{}
+	checkParams.SetVerificationSid(verificationID)
+	checkParams.SetCode(code)
+	resp, err := client.VerifyV2.CreateVerificationCheck(t.VerifyServiceSID, checkParams)
+	if err != nil || resp.Status == nil || *resp.Status != "approved" {
```

Comment:
What do you think about separating out some of the other error statuses? The two I'm thinking of are `max_attempts_reached` and `expired` - both could be helpful to surface to the client for enabling/disabling the input and "resend-code" button.

Reply:
👍  Great input. Will have a look at it.

Reply:
Yes, would definitely be nice to know

Reply:
done, but the twilio verification api might always response with just an error in those cases: https://www.twilio.com/docs/verify/api/verification-check#check-a-verification